src/darwin/Framework/CHIP/MTRDefines.h

```diff
@@ -37,13 +37,36 @@
 
 #pragma mark - Deprecation macros (can be overriden via build system)
 
+/**
+ * MTR_SKIP_AVAILABILITY_ANNOTATIONS can be used to turn off availability
+ * annotations, to allow compiling a version of Matter.framework for "local
+ * use", not as a system framework.
+ */
+#if !defined(MTR_SKIP_AVAILABILITY_ANNOTATIONS)
+#define MTR_SKIP_AVAILABILITY_ANNOTATIONS 0
+#endif
+
 #ifndef MTR_DEPRECATED
```

Comment:
Do we actually need to allow these to be overridden from outside? Otherwise we could get rid off the `ifndef` checks and simplify things a bit into a single `#if MTR_SKIP_AVAILABILITY_ANNOTATIONS`

Reply:
> Do we actually need to allow these to be overridden from outside

Good point.  I don't think we do; I am not sure why we set things up this way.  Will do the simplification.

Reply:
And we can nix the conditionality on MTR_NEWLY_DEPRECATED/AVAILABLE too.